### PR TITLE
Added support for log file headers

### PR DESCRIPTION
**Pull request by tmsrtx**
Added property `initialFileContents` to `DDLogFileManagerDefault` to support headers in log files.
Override implementation of `initialFileContents` to insert header text at the beginning of every log file created by the `createNewLogFile:` method.

### New Pull Request Checklist

* [x] I have read and understood the [CONTRIBUTING guide](https://github.com/CocoaLumberjack/CocoaLumberjack/blob/master/.github/CONTRIBUTING.md)
* [x] I have read the [Documentation](http://cocoadocs.org/docsets/CocoaLumberjack/)
* [x] I have searched for a similar pull request in the [project](https://github.com/CocoaLumberjack/CocoaLumberjack/pulls) and found none

* [x] I have updated this branch with the latest master to avoid conflicts (via merge from master or rebase)
* [x] I have added the required tests to prove the fix/feature I am adding
* [x] I have updated the documentation (if necessary)
* [x] I have run the tests and they pass
* [x] I have run the lint and it passes (`pod lib lint`)

This merge request fixes / refers to the following issues: ...

### Pull Request Description

We wish to include the version and build date of our application at the beginning of the log files generated by CocoaLumberjack. I have not been able to find any obvious way of doing this so I propose adding an `initialFileContents` property to `DDLogFileManagerDefault` which the  `createNewLogFile:` method then uses as an argument for the 'contents' parameter (instead of nil) when calling `createFileAtPath:contents:attributes:`

The intention is to achieve something which is possible in NLog (though NLog is a very different beast), using `LayoutWithHeaderAndFooter`.

### CODE REVIEW

**Comment by rivera-ernesto** [comment]
Looks good, but I would rename `initialFileContents` with something like `logFileHeader`.

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,6 +22,7 @@
 - DDFileLogger log message is overridden #924
 - Fix thread safety issue in DDFileLogger #986
 - Fix availability checks and memory leak #996
+- Added `initialFileContents` property to `DDLogFileManagerDefault` to set log file header
```
Comment (rivera-ernesto) [inline CHANGELOG.md]: One last left here and we're good to merge.

Comment (tmsrtx) [reply CHANGELOG.md]: Oh, right. Forgot that one, thanks :-)
I'm also implementing some test cases so both OSSPolice and codecov will hopefully be happy.

File: Tests/Tests/DDLogFileManagerTests.m [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+@implementation DDLogFileManagerDefaultTests
+
+- (void)setUp {
+    [super setUp];
+    [self setUpLogFileManager];
+}
+
+- (void)setUpLogFileManager {
+    NSString *logsDirectory = NSTemporaryDirectory();
```
Comment (lolgear) [inline Tests/Tests/DDLogFileManagerTests.m]: Please, use `__auto_type` instead.

Tests are creatures from the modern world.

Comment (tmsrtx) [reply Tests/Tests/DDLogFileManagerTests.m]: Could you please explain to me what the benefit of using `__auto_type` in this case is?
It uses up more characters, does not explicitly tell a reader what the type is and, I assume, the compiled code will be the same. I can understand wanting to use `__auto_type` to infer the type when the resulting type is complex but here it is very simple.
I don't write Objective-C on an everyday basis so I'm just being curious here :-)

Comment (lolgear) [reply Tests/Tests/DDLogFileManagerTests.m]: @tmsrtx 

It is not the only one place, I just choose random place.

You have more "simple type" example.

```objective-c
        unsigned long long fileSize = [[[NSFileManager defaultManager] attributesOfItemAtPath:filePath error:nil][NSFileSize] unsignedLongLongValue];
```

```objective-c
        __auto_type fileSize = [[[NSFileManager defaultManager] attributesOfItemAtPath:filePath error:nil][NSFileSize] unsignedLongLongValue];
```

And, of course, writing mantras like: `NSString *something = [[NSString alloc] init..]` is not the best style.

What about top-level functions and this example?

```objective-c
    NSString *logsDirectory = NSTemporaryDirectory();
```

It is already incorrect. 

```objective-c
NSTemporaryDirectory(); // return type nonnull NSString *
```

For example, if `nullability` can become a part of type ( or any type placeholder construction ), my advise will take zero efforts.

**Comment by lolgear** [comment]
However, adding "Header" for file logger is really discouraged.

It is not the point for a file logger.
File logger can gather logs into a file. Ok, it is a content of a file.
Next, if you like to post-process file, you can ( before archiving or sending to your server ) append header, footer, any type of stars, whistles, bells etc. You can also add your best poem or quote of the day.

**Comment by rivera-ernesto** [comment]
I see what you say @lolgear , but at the same time by default it will be empty. So it looks good to me.

File: Classes/DDFileLogger.m [hunk 3]
```diff
@@ -427,9 +429,28 @@ - (NSString *)newLogFileName {
...
+
+- (NSData *)logFileHeaderData {
+    NSString *fileHeaderStr = [self logFileHeader];
+    
+    if (fileHeaderStr == nil || [fileHeaderStr length] == 0) {
+        return nil;
+    }
+    
+    // Ensure that we have a newline at the end of the string
+    fileHeaderStr = [NSString stringWithFormat:@"%@\n", fileHeaderStr];
```
Comment (hhanesand) [inline Classes/DDFileLogger.m]:
```suggestion
    unichar lastChar = [fileHeaderStr characterAtIndex:fileHeaderStr.length - 1];
    if (![[NSCharacterSet newlineCharacterSet] characterIsMember:lastChar]) {
        fileHeaderStr = [NSString stringWithFormat:@"%@\n", fileHeaderStr];
    }
```

**Pull request merged**